lib/contentful-util.js

```diff
@@ -62,22 +62,29 @@ function normalizeEntry({ contentTypes, entry, options }) {
     return Object.assign({}, fields, { __metadata: metadata });
 }
 
-function resolveLinks(entry, assets, entries) {
-    if (!entry.fields) return entry;
+function resolveLink(value, assets, entries) {
+    if (Array.isArray(value)) {
+        return value.map(childValue => resolveLink(childValue, assets, entries));
+    }
 
-    const resolvedFields = Object.keys(entry.fields).reduce((result, fieldName) => {
-        const localeKey = Object.keys(entry.fields[fieldName])[0];
+    if (value.sys && value.sys.type === 'Link') {
+        const searchPool = value.sys.linkType === 'Asset' ? assets : entries;
+        const match = searchPool.find(item => item.sys && item.sys.id === value.sys.id);
 
-        let value = entry.fields[fieldName][localeKey];
+        if (match) {
+            return resolveLinksInEntry(match, assets, entries);
+        }
+    }
 
-        if (value.sys && value.sys.type === 'Link') {
-            const searchPool = value.sys.linkType === 'Asset' ? assets : entries;
-            const match = searchPool.find(item => item.sys && item.sys.id === value.sys.id);
+    return value;
```

Comment:
we just had a bug in stackbit-pull related to this:
https://www.notion.so/stackbit/Container-User-Bug-stackbit-pull-encode-crash-2c208d26d8924ea591c9e6048a3ec43b

In short, if linked object is not found, then it is better to return `null`:
```js
if (match) {
  return resolveLinksInEntry(match, assets, entries);
} else {
  return null;
}
```

Otherwise, we end up with CMS specific link object inside result data:
```json
{
  "sys": {
    "type": "Link",
    "linkType": "...",
    "id": "..."
  }
}
```
And SSG logic is not expecting to get such an object.

Same should apply for sanity and other source plugins.

Reply:
Updated in https://github.com/stackbithq/sourcebit-source-contentful/pull/3/commits/b51f2fc8385746e8e4da386bb18050545ff1c6e6.